v4/export/sql.go

```diff
@@ -50,7 +50,9 @@ func ShowCreateDatabase(db *sql.Conn, database string) (string, error) {
 	query := fmt.Sprintf("SHOW CREATE DATABASE `%s`", escapeString(database))
 	err := simpleQuery(db, query, handleOneRow)
 	if err != nil {
-		return "", errors.Annotatef(err, "sql: %s", query)
+		// Falling back to simple create statement. This can happen
+		// with MemSQL/SingleStore as MemSQL doesn't support `SHOW CREATE DATABASE`
```

Comment:
can the fallback be restricted to only the `SHOW CREATE DATABASE` unsupported error (e.g. detecting if the error code is 1064 (You have an error in your SQL syntax))?

Reply:
Restricted the fallback to error code 1706:
```ERROR 1706 (HY000): Feature 'SHOW CREATE DATABASE' is not supported by MemSQL.```